### PR TITLE
[expo-updates] Make Updates API work with native debug

### PR DESCRIPTION
**Pull request by douglowder**
# Why

If an app is using `expo-updates` and is built with the `EX_UPDATES_NATIVE_DEBUG` environment variable set to allow native debugging, updates work correctly on the native side, and download and run automatically as expected. However, some methods in the Updates API (`checkForUpdateAsync()`, `fetchUpdateAsync()`, `reloadAsync()`) are currently implemented to only work in release builds.

# How

Expose a new `nativeDebug` property in the Updates constants, and check for that in the method implementations.

# Test Plan

Added a custom `App.js` in the app created by `create-update-test.js` that allows these methods to be exercised.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by wschurman — APPROVED** [review]

File: packages/expo-updates/src/Updates.ts [hunk 1]
```diff
@@ -132,7 +132,7 @@ export async function reloadAsync(): Promise<void> {
   if (!ExpoUpdates.reload) {
     throw new UnavailabilityError('Updates', 'reloadAsync');
   }
-  if (__DEV__ && !isUsingExpoDevelopmentClient) {
+  if (!ExpoUpdates?.nativeDebug && (__DEV__ || isUsingExpoDevelopmentClient)) {
```
Comment (wodin) [inline packages/expo-updates/src/Updates.ts:135]: @douglowder, before this change, if `isUsingExpoDevelopmentClient` was true, then the exception would not be thrown and the `reloadAsync()` call would succeed.

After the change, the exception will be thrown if `isUsingExpoDevelopmentClient` is true, unless `nativeDebug` is true.

Was this intended?

There are users who were using `reloadAsync()` for restarting the app for purposes other than OTA updates (e.g. after changing language settings). After this change, the dev client can't be restarted, unless `nativeDebug` is enabled.

https://forums.expo.dev/t/i-cant-restart-the-app-using-expo-updates-to-change-language/74237?u=wodin

Comment (douglowder) [reply packages/expo-updates/src/Updates.ts:135]: Good catch! I'll get this fixed.

Comment (wodin) [reply packages/expo-updates/src/Updates.ts:135]: Thanks! I've posted a followup to the thread on the forums.

Comment (afj95) [reply packages/expo-updates/src/Updates.ts:135]: @douglowder
It is working very well with me thanks to @wodin.

Comment (douglowder) [reply packages/expo-updates/src/Updates.ts:135]: PR is up at https://github.com/expo/expo/pull/24588

**Pull request merged**